types/params.go

```diff
@@ -109,6 +110,20 @@ type SynchronyParams struct {
 	MessageDelay time.Duration `json:"message_delay,string"`
 }
 
+/*
+AdaptiveSynchronyParams ensures an exponential backoff for timestamp validation by relaxing MessageDelay by a factor of
+10% each subsequent round a proposal's timeliness is calculated. This is meant to facilitate the progression of
+consensus if bad synchrony parameters are set or become insufficient to preserve liveness.
+
+Where r is the consensus round, MSGDELAY(r) == MSGDELAY(0) * (1.1)^r.
+*/
+func AdaptiveSynchronyParams(precision time.Duration, messageDelay time.Duration, round int32) SynchronyParams {
```

Comment:
Why we receive the two fields of `SynchronyParams` here, instead of an instance of it?

Reply:
Felt cleaner and more efficient to implement a constructor instead of creating synch params and needing to ensure the message delay value had been properly adapted either before or after.

Reply:
@glnro I think what @cason means is that you could pass just the original `SynchronyParams` in one simgle param,  so that all callsites of `AdaptiveSynchronyParams` can be simplified. Something like this:
```go
func AdaptiveSynchronyParams(origParams SynchronyParams, round int32) SynchronyParams {
...
}
```
And then, inside the function, you can still keep the constructor as you have it now.

Reply:
Actually I disagree and the more I think about it, the more I think we should not be able to create synchrony params and pass them around without ensuring that the round is accounted for. Wdyt about making `synchronyParams` unexported at the package level and only having a public constructor that additionally takes a round.

Reply:
Fair enough. Then I think the best solution is what you have at present.
@cason wdyt?